core/src/main/java/org/apache/calcite/rel/mutable/MutableScan.java

```diff
@@ -36,11 +36,12 @@ public static MutableScan of(TableScan scan) {
   @Override public boolean equals(Object obj) {
     return obj == this
         || obj instanceof MutableScan
-        && rel.equals(((MutableScan) obj).rel);
+            && rel.getTable().getQualifiedName().equals(((MutableScan) obj).rel
+                .getTable().getQualifiedName());
```

Comment:
I will remove this change once https://github.com/apache/calcite/pull/1531/files gets merged.

Reply:
Can you note that dependency in the JIRA cases.

Reply:
Thanks ~ 
That is https://issues.apache.org/jira/browse/CALCITE-3447